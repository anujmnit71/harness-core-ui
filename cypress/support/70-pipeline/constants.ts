/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const projectId = 'project1'
const accountId = 'accountId'
const orgIdentifier = 'default'

export const pipelineListAPI = `/pipeline/api/pipelines/list?routingId=${accountId}&accountIdentifier=${accountId}&projectIdentifier=${projectId}&module=cd&orgIdentifier=${orgIdentifier}&searchTerm=&page=0&sort=lastUpdatedAt%2CDESC&size=20`
export const pipelineSummaryAPI = `/pipeline/api/pipelines/summary/appdtest?routingId=${accountId}&accountIdentifier=${accountId}&orgIdentifier=${orgIdentifier}&projectIdentifier=${projectId}`
export const pipelineDetailsAPI = `/pipeline/api/pipelines/appdtest?accountIdentifier=${accountId}&orgIdentifier=${orgIdentifier}&projectIdentifier=${projectId}`
export const pipelineExecutionSummaryAPI = `/pipeline/api/pipelines/execution/summary?routingId=${accountId}&accountIdentifier=${accountId}&projectIdentifier=${projectId}&orgIdentifier=${orgIdentifier}&module=cd&size=20&pipelineIdentifier=*&page=0&myDeployments=false`
export const pipelineExecutionAPI = `/pipeline/api/pipelines/execution/C9mgNjxSS7-B-qQek27iuA?routingId=${accountId}&orgIdentifier=${orgIdentifier}&projectIdentifier=${projectId}&accountIdentifier=${accountId}`
export const pipelineExecutionForNodeAPI = `/pipeline/api/pipelines/execution/C9mgNjxSS7-B-qQek27iuA?routingId=${accountId}&orgIdentifier=${orgIdentifier}&projectIdentifier=${projectId}&accountIdentifier=${accountId}&stageNodeId=g_LkakmWRPm-wC6rfC2ufg`

export const pipelinesListCallResponse = {
  status: 'SUCCESS',
  data: {
    content: [
      {
        name: 'Parallel Pipelines',
        identifier: 'Parallel_Pipelines',
        tags: {},
        version: 33,
        numOfStages: 7,
        createdAt: 1643729398277,
        lastUpdatedAt: 1643729619074,
        modules: ['cd'],
        executionSummaryInfo: {
          numOfErrors: [0, 0, 0, 0, 0, 0, 15],
          deployments: [0, 0, 0, 0, 0, 0, 15],
          lastExecutionTs: 1643739804626,
          lastExecutionStatus: 'Failed',
          lastExecutionId: '-_gYM5jES1Wnow2BKbvDBQ'
        },
        filters: {
          cd: {
            deploymentTypes: ['Kubernetes'],
            environmentNames: ['prod environment'],
            serviceNames: ['Normal Service'],
            infrastructureTypes: ['KubernetesDirect']
          }
        },
        stageNames: ['Stage1', 'Stage2', 'Stage3', 'Stage4', 'Stage5', 'Stage6', 'Stage7'],
        gitDetails: {
          objectId: null,
          branch: null,
          repoIdentifier: null,
          rootFolder: null,
          filePath: null,
          repoName: null
        },
        entityValidityDetails: {
          valid: true,
          invalidYaml: null
        }
      },
      {
        name: 'ParallStage',
        identifier: 'ParallStage',
        tags: {},
        version: 154,
        numOfStages: 6,
        createdAt: 1641473199732,
        lastUpdatedAt: 1643184469790,
        modules: ['cd', 'ci'],
        executionSummaryInfo: {
          numOfErrors: [1, 0, 0, 0, 0, 0, 0],
          deployments: [6, 0, 20, 0, 0, 9, 0],
          lastExecutionTs: 1643601881783,
          lastExecutionStatus: 'Aborted',
          lastExecutionId: 'wjWjWYaBRNeT1Rrza4N9Dg'
        },
        filters: {
          cd: {
            deploymentTypes: ['Kubernetes'],
            environmentNames: [
              'test',
              'Devsf',
              'Very Long Env asd asd asd Kapil',
              'fdsggg',
              'asdasd',
              'prod environment'
            ],
            serviceNames: ['Newdds', 'test', 'Very Big Service Name Really', 'web', 'Normal Service', 'gfgfgf asdsad'],
            infrastructureTypes: ['KubernetesDirect']
          },
          ci: {
            repoNames: ['CCD']
          }
        },
        stageNames: ['Stage 5', 'Stage 6', 'Stage 3', 'Stage 4', 'Stage 1', 'Stage 2'],
        gitDetails: {
          objectId: null,
          branch: null,
          repoIdentifier: null,
          rootFolder: null,
          filePath: null,
          repoName: null
        },
        entityValidityDetails: {
          valid: true,
          invalidYaml: null
        }
      },
      {
        name: 'Pipeline 1911',
        identifier: 'Pipeline_1911',
        tags: {
          tag1: ''
        },
        version: 14,
        numOfStages: 1,
        createdAt: 1637310118606,
        lastUpdatedAt: 1640093470460,
        modules: ['cd'],
        executionSummaryInfo: {
          numOfErrors: [],
          deployments: []
        },
        filters: {
          cd: {
            deploymentTypes: ['Kubernetes'],
            environmentNames: ['env 2'],
            serviceNames: ['service 4'],
            infrastructureTypes: ['KubernetesDirect']
          }
        },
        stageNames: ['s1'],
        gitDetails: {
          objectId: '40d8e60d7e4b46f1a41deba2af7ee9c324726c13',
          branch: 'main',
          repoIdentifier: 'Git_Sync_Cypress',
          rootFolder: 'cypressQAResources/.harness/',
          filePath: 'Pipeline_1911.yaml',
          repoName: null
        },
        entityValidityDetails: {
          valid: false,
          invalidYaml:
            'pipeline:\n    name: Pipeline 1911\n    identifier: Pipeline_1911\n    projectIdentifier: Cypress_Git_Sync_Saas\n    orgIdentifier: default\n    tags:\n        tag1: ""\n    stages:\n        - stage:\n              name: s1\n              identifier: s1\n              description: ""\n              type: Deployme\n              spec:\n                  serviceConfig:\n                      serviceRef: service_4\n                      serviceDefinition:\n                          type: Kubernetes\n                          spec:\n                              variables: []\n                              manifests:\n                                  - manifest:\n                                        identifier: test\n                                        type: HelmChart\n                                        spec:\n                                            store:\n                                                type: Http\n                                                spec:\n                                                    connectorRef: account.puthrayahelm\n                                            chartName: test do\n                                            chartVersion: <+input>\n                                            helmVersion: V2\n                                            skipResourceVersioning: false\n                  infrastructure:\n                      environmentRef: env_2\n                      infrastructureDefinition:\n                          type: KubernetesDirect\n                          spec:\n                              connectorRef: <+input>\n                              namespace: <+input>\n                              releaseName: release-<+INFRA_KEY>\n                      allowSimultaneousDeployments: false\n                  execution:\n                      steps:\n                          - step:\n                                type: ShellScript\n                                name: test1\n                                identifier: test1\n                                spec:\n                                    shell: Bash\n                                    onDelegate: true\n                                    source:\n                                        type: Inline\n                                        spec:\n                                            script: <+input>\n                                    environmentVariables: []\n                                    outputVariables: []\n                                    executionTarget: {}\n                                timeout: 10m\n                      rollbackSteps: []\n              tags: {}\n              failureStrategies:\n                  - onFailure:\n                        errors:\n                            - AllErrors\n                        action:\n                            type: StageRollback\n    variables:\n        - name: var2411\n          type: Number\n          value: ""\n'
        }
      }
    ],
    pageable: {
      sort: {
        sorted: true,
        unsorted: false,
        empty: false
      },
      pageNumber: 0,
      pageSize: 20,
      offset: 0,
      paged: true,
      unpaged: false
    },
    last: false,
    totalPages: 1,
    totalElements: 3,
    first: true,
    sort: {
      sorted: true,
      unsorted: false,
      empty: false
    },
    numberOfElements: 3,
    size: 20,
    number: 0,
    empty: false
  },
  metaData: null,
  correlationId: 'f8b4e248-f6ba-47eb-938f-b205e6de5d4b'
}
