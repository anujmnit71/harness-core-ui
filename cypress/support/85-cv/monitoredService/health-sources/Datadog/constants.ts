/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const projectIdentifier = 'project1'
const orgIdentifier = 'default'
const accountId = 'accountId'
const dataSourceType = 'DATADOG_METRICS'
const selectedDashboardId = 'gas-53u-ruk'
const configuredMetric = 'datadog.agent.python.version'
export const connectorIdentifier = 'datadog'
export const selectedDashboardName = "praveen's Timeboard 5 Apr 2019 18:16"

export const dashboards = {
  dashboardsAPI: `/cv/api/datadog-metrics/dashboards?routingId=${accountId}&accountId=${accountId}&projectIdentifier=${projectIdentifier}&orgIdentifier=${orgIdentifier}&connectorIdentifier=${connectorIdentifier}&pageSize=7&offset=0&tracingId=*`,
  dashboardsResponse: {
    status: 'SUCCESS',
    data: {
      totalPages: 2,
      totalItems: 14,
      pageItemCount: 7,
      pageSize: 7,
      content: [
        { id: 'a97-8cs-426', name: 'test', path: '/dashboard/a97-8cs-426/test' },
        {
          id: 'c8p-tpb-fn4',
          name: "praveen's Screenboard 11 Apr 2019 17:46",
          path: '/dashboard/c8p-tpb-fn4/praveens-screenboard-11-apr-2019-1746'
        },
        {
          id: 'iph-s8x-tu9',
          name: "pranjal's Timeboard 7 May 2019 15:32",
          path: '/dashboard/iph-s8x-tu9/pranjals-timeboard-7-may-2019-1532'
        },
        {
          id: 'gas-53u-ruk',
          name: "praveen's Timeboard 5 Apr 2019 18:16",
          path: '/dashboard/gas-53u-ruk/praveens-timeboard-5-apr-2019-1816'
        },
        {
          id: 'vxi-z6n-udt',
          name: "praveen's Timeboard 20 Dec 2018 12:25",
          path: '/dashboard/vxi-z6n-udt/praveens-timeboard-20-dec-2018-1225'
        },
        {
          id: '8i3-d5k-7wq',
          name: "parnian's Timeboard 18 Jul 2018 15:05",
          path: '/dashboard/8i3-d5k-7wq/parnians-timeboard-18-jul-2018-1505'
        },
        {
          id: 'jgw-pbc-evf',
          name: "praveen's Timeboard 18 Jul 2018 15:01",
          path: '/dashboard/jgw-pbc-evf/praveens-timeboard-18-jul-2018-1501'
        }
      ],
      pageIndex: 0,
      empty: false
    },
    metaData: null,
    correlationId: 'eafd4a53-f439-4202-91ae-6d59efa2195c'
  }
}

export const metrics = {
  getMetricsCall: `/cv/api/metric-pack?routingId=${accountId}&projectIdentifier=${projectIdentifier}&orgIdentifier=${orgIdentifier}&accountId=${accountId}&dataSourceType=${dataSourceType}`,
  getMetricsResponse: {
    metaData: {},
    resource: [
      {
        uuid: '5r4EUPHhRFaXtb6L7IHAkA',
        accountId: '${accountId}',
        orgIdentifier,
        projectIdentifier: 'Harshil',
        dataSourceType,
        identifier: 'Errors',
        category: 'Errors',
        metrics: [
          {
            name: 'Errors',
            type: 'ERROR',
            path: null,
            validationPath: null,
            responseJsonPath: null,
            validationResponseJsonPath: null,
            thresholds: [],
            included: false
          }
        ],
        thresholds: null
      },
      {
        uuid: 'MGTNKXwYTrGfC27sch_u4Q',
        accountId: '${accountId}',
        orgIdentifier,
        projectIdentifier: 'Harshil',
        dataSourceType,
        identifier: 'Infrastructure',
        category: 'Infrastructure',
        metrics: [
          {
            name: 'Infrastructure',
            type: 'INFRA',
            path: null,
            validationPath: null,
            responseJsonPath: null,
            validationResponseJsonPath: null,
            thresholds: [],
            included: false
          }
        ],
        thresholds: null
      },
      {
        uuid: 'yqsrmlaBR-2Y4oawK9iYaQ',
        accountId: '${accountId}',
        orgIdentifier,
        projectIdentifier: 'Harshil',
        dataSourceType,
        identifier: 'Performance',
        category: 'Performance',
        metrics: [
          {
            name: 'Throughput',
            type: 'THROUGHPUT',
            path: null,
            validationPath: null,
            responseJsonPath: null,
            validationResponseJsonPath: null,
            thresholds: [],
            included: false
          },
          {
            name: 'Response Time',
            type: 'RESP_TIME',
            path: null,
            validationPath: null,
            responseJsonPath: null,
            validationResponseJsonPath: null,
            thresholds: [],
            included: false
          },
          {
            name: 'Other',
            type: 'ERROR',
            path: null,
            validationPath: null,
            responseJsonPath: null,
            validationResponseJsonPath: null,
            thresholds: [],
            included: false
          }
        ],
        thresholds: null
      }
    ],
    responseMessages: []
  }
}

export const metricTags = {
  getMetricsTags: `/cv/api/datadog-metrics/metric-tags?routingId=${accountId}&projectIdentifier=${projectIdentifier}&orgIdentifier=${orgIdentifier}&accountId=${accountId}&connectorIdentifier=${connectorIdentifier}&tracingId=*&metric=datadog.agent.python.version`,
  getMetricsTagsResponse: {
    status: 'SUCCESS',
    data: ['agent_version_major:7'],
    metaData: null,
    correlationId: 'd5055d0d-e6d2-4ac5-8f29-c373ad9ae893'
  }
}

export const activeMetrics = {
  getActiveMetrics: `/cv/api/datadog-metrics/active-metrics?routingId=${accountId}&projectIdentifier=${projectIdentifier}&orgIdentifier=${orgIdentifier}&accountId=${accountId}&connectorIdentifier=${connectorIdentifier}&tracingId=*`,
  getActiveMetricsResponse: {
    status: 'SUCCESS',
    data: [configuredMetric],
    metaData: null,
    correlationId: 'eb9d6780-8db5-4f51-b476-aac2f63a00d0'
  }
}

export const dashboardDetails = {
  getDashboardDetails: `/cv/api/datadog-metrics/dashboard-details?routingId=${accountId}&projectIdentifier=${projectIdentifier}&orgIdentifier=${orgIdentifier}&accountId=${accountId}&connectorIdentifier=${connectorIdentifier}&tracingId=*&dashboardId=${selectedDashboardId}&path=${selectedDashboardId}`,
  getDashboardDetailsResponse: {
    status: 'SUCCESS',
    data: [
      {
        widgetName: configuredMetric,
        dataSets: [
          {
            name: `avg:${configuredMetric}{kube_service:cv24-7-test-dev-cv-todolist-cv-24-7-appd}`,
            query: `avg:${configuredMetric}{kube_service:cv24-7-test-dev-cv-todolist-cv-24-7-appd}`
          }
        ]
      }
    ],
    metaData: null,
    correlationId: 'd480b740-f160-48d7-910d-3491c557ff40'
  }
}

export const dataDogMonitoredService = {
  status: 'SUCCESS',
  data: {
    createdAt: 1644978191147,
    lastModifiedAt: 1644978191147,
    monitoredService: {
      orgIdentifier,
      projectIdentifier: 'Harshil',
      identifier: 'service1_env1',
      name: 'service1_env1',
      type: 'Application',
      description: '',
      serviceRef: 'service1',
      environmentRef: 'env1',
      environmentRefList: ['env1'],
      tags: {},
      sources: {
        healthSources: [
          {
            name: 'DD new',
            identifier: 'DD_new',
            type: 'DatadogMetrics',
            spec: {
              connectorRef: 'datadog',
              feature: 'Datadog Cloud Metrics',
              metricDefinitions: [
                {
                  identifier: 'DD metric',
                  metricName: 'DD metric',
                  riskProfile: { category: 'Errors', metricType: 'ERROR', thresholdTypes: ['ACT_WHEN_HIGHER'] },
                  analysis: {
                    liveMonitoring: { enabled: false },
                    deploymentVerification: {
                      enabled: true,
                      serviceInstanceFieldName: '',
                      serviceInstanceMetricPath: null
                    },
                    riskProfile: { category: 'Errors', metricType: 'ERROR', thresholdTypes: ['ACT_WHEN_HIGHER'] }
                  },
                  sli: { enabled: false },
                  dashboardId: null,
                  dashboardName: 'group-1',
                  metricPath: 'DD metric',
                  query: `${configuredMetric}{}.rollup(avg, 60)`,
                  groupingQuery: `${configuredMetric}{}.rollup(avg, 60)`,
                  metric: configuredMetric,
                  aggregation: null,
                  serviceInstanceIdentifierTag: '',
                  metricTags: [''],
                  isManualQuery: false,
                  isCustomCreatedMetric: true
                }
              ]
            }
          }
        ],
        changeSources: [
          {
            name: 'Harness CD Next Gen',
            identifier: 'harness_cd_next_gen',
            type: 'HarnessCDNextGen',
            enabled: true,
            spec: {},
            category: 'Deployment'
          }
        ]
      },
      dependencies: []
    }
  },
  metaData: null,
  correlationId: '88174d93-852a-4ff5-9dce-60b5e43d27cb'
}
