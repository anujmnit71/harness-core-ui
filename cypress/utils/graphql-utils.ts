export const hasOperationName: (req: any, operationName: string) => boolean = (req, operationName) => {
  const { body } = req
  // eslint-disable-next-line no-prototype-builtins
  return body.hasOwnProperty('operationName') && body.operationName === operationName
}

export const aliasQuery: (req: any, operationName: string) => void = (req, operationName) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`
  }
}
