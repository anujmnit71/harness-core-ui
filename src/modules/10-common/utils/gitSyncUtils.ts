/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitSyncConfig } from 'services/cd-ng'
import { folderPathName, yamlFileExtension } from './StringUtils'

export const getRepoDetailsByIndentifier = (identifier: string | undefined, repos: GitSyncConfig[]) => {
  return repos.find((repo: GitSyncConfig) => repo.identifier === identifier)
}

export const validateFilePath = (filePath: string): boolean => {
  if (!filePath.endsWith(yamlFileExtension)) {
    return false
  } else {
    const fullPath = filePath.slice(0, filePath.length - yamlFileExtension.length)
    const subPaths = fullPath.split('/')
    return subPaths.every((folderName: string) => {
      return folderName.length && folderName.match(folderPathName)?.length
    })
  }
}
