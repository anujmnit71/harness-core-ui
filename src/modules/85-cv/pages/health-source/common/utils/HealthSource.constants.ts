/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const HealthSourceFieldNames = {
  MANUAL_QUERY: 'isManualQuery',
  IDENTIFIER: 'identifier',
  METRIC_NAME: 'metricName',
  METRIC_DATA: 'metricData',
  GROUP_NAME: 'groupName',
  CONTINUOUS_VERIFICATION: 'continuousVerification',
  HEALTH_SCORE: 'healthScore',
  SLI: 'sli',
  RISK_CATEGORY: 'riskCategory',
  HIGHER_BASELINE_DEVIATION: 'higherBaselineDeviation',
  LOWER_BASELINE_DEVIATION: 'lowerBaselineDeviation'
}
