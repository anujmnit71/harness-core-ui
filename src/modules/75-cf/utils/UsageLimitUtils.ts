/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactNode } from 'react'
import { BannerType } from '@common/layouts/Constants'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import { formatToCompactNumber } from '@cf/utils/CFUtils'
import type { UseStringsReturn } from 'framework/strings'

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  monthlyActiveUsers: CheckFeatureReturn,
  additionalLicenseProps: Record<string, any>
): ReactNode => {
  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps

  const clientMauUsageCount = Number(monthlyActiveUsers?.featureDetail?.count)
  const clientMauPlanLimit = Number(monthlyActiveUsers?.featureDetail?.limit)

  const clientMauUsagePercentage = Math.trunc((clientMauUsageCount / clientMauPlanLimit) * 100)
  const clientMauPlanLimitFormatted = formatToCompactNumber(clientMauPlanLimit)

  const showInfoBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 90 && clientMauUsagePercentage < 100
  const showWarningBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 100

  let message!: string
  let bannerType!: BannerType

  if (isFreeEdition) {
    if (!showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.freePlan.approachingLimit', {
            clientMauUsagePercentage,
            clientMauPlanLimitFormatted
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: getString('cf.planEnforcement.freePlan.upgradeRequired', {
          clientMauPlanLimitFormatted
        }),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  if (isTeamEdition || isEnterpriseEdition) {
    if (showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.teamEnterprisePlan.approachingLimit', {
            clientMauUsagePercentage
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: () => getString('cf.planEnforcement.teamEnterprisePlan.upgradeRequired'),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  return {
    message,
    bannerType
  }
}
