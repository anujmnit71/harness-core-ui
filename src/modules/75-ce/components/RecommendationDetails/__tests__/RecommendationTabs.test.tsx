/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'

import RecommendationTabs from '../RecommendationTabs'
import { RecommendationType } from '../constants'

describe('test cases for recommendation tabs', () => {
  test('should be able to render recommendation tabs', async () => {
    const { container } = render(
      <TestWrapper>
        <RecommendationTabs
          selectedRecommendation={RecommendationType.CostOptimized}
          setSelectedRecommendation={jest.fn()}
          setCPUReqVal={jest.fn()}
          setMemReqVal={jest.fn()}
          setMemLimitVal={jest.fn()}
          costOptimizedSavings={100}
          performanceOptimizedSavings={200}
          currentSavings={50}
          isPerfOptimizedCustomized={false}
          isCostOptimizedCustomized={false}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
