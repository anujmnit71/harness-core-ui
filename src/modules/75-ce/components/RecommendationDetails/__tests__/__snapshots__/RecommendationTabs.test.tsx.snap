// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for recommendation tabs should be able to render recommendation tabs 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main recommendationTypeContainer StyledProps--padding-small"
  >
    <div
      class="Layout--horizontal StyledProps--main recommendations"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main recommendationTypeText selectedTab"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="costOptimized"
        >
          <p
            class="StyledProps--font StyledProps--main selectedTab StyledProps--font-size-small StyledProps--font-align-center"
          >
            ce.recommendation.detailsPage.costOptimized
          </p>
        </div>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main recommendationTypeText"
      >
        <div
          class="Text--withDocsTooltip"
          data-tooltip-id="performanceOptimized"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-align-center"
          >
            ce.recommendation.detailsPage.performanceOptimized
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
